### PR TITLE
CCM-5883: Set up Amplify backend

### PR DESCRIPTION
**Pull request by chris-elliott-nhsd**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/workflows/cicd-deploy-amplify-branch.yaml [hunk 1]
```diff
@@ -9,6 +9,11 @@ on:
         required: true
         default: dev
         type: string
+      backendBranch:
+        description: Name of the branch to use for Amplify backend
+        required: true
+        default: feature/CCM-5615_amplify-backend
```
Comment (chris-elliott-nhsd) [inline .github/workflows/cicd-deploy-amplify-branch.yaml:15]: change this to 'main' before merge

File: amplify/auth/resource.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { defineAuth } from '@aws-amplify/backend';
+
+export const auth = defineAuth({
```
Comment (mark-r-bjss) [inline amplify/auth/resource.ts:3]: Why do we need this?

File: public/lib/nhsuk-8.1.1.min.js
Comment (mark-r-bjss) [inline public/lib/nhsuk-8.1.1.min.js:1]: This should not change, do we need to exclude it from linting/formatting?

File: public/lib/nhsuk.min.js
Comment (mark-r-bjss) [inline public/lib/nhsuk.min.js:1]: This should not change, do we need to exclude it from linting/formatting?

File: amplify/auth/resource.ts (hunk 2 above)
Comment (chris-elliott-nhsd) [reply amplify/auth/resource.ts:3]: `auth` is a required field for defining the Amplify backend - it's not required by the typescript definitions but the deployment fails if it isn't there

File: src/styles/nhsuk-8.1.1.min.css
Comment (mark-r-bjss) [inline src/styles/nhsuk-8.1.1.min.css:1]: This should not change, do we need to exclude it from linting/formatting?

File: public/lib/nhsuk-8.1.1.min.js
Comment (chris-elliott-nhsd) [reply public/lib/nhsuk-8.1.1.min.js:1]: reverted and updated .editorconfig to not check .min.js or .min.css files for final newlines

File: public/lib/nhsuk.min.js
Comment (chris-elliott-nhsd) [reply public/lib/nhsuk.min.js:1]: reverted and updated .editorconfig to not check .min.js or .min.css files for final newlines

File: scripts/create_sandbox.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+set -euo pipefail
+
+echo "Creating sandbox"
+
+npm run create-sandbox
+
+# wait for Amplify outputs file to be available
+wait_seconds=0
+max_wait_seconds=600
+wait_interval=10
```
[the remark this answers is not in the document]
Comment (chris-elliott-nhsd) [reply scripts/create_sandbox.sh]: updated

File: src/styles/nhsuk-8.1.1.min.css
Comment (chris-elliott-nhsd) [reply src/styles/nhsuk-8.1.1.min.css:1]: reverted and updated .editorconfig to not check .min.js or .min.css files for final newlines

**Review by mark-r-bjss — APPROVED** [review]

File: amplify_outputs.json
Comment (bhansell1) [inline amplify_outputs.json:1]: Does this file need to be checked in? It's modified when a sandbox is created. Seems really easy to accidentally check this file in with all the content.

Comment (chris-elliott-nhsd) [reply amplify_outputs.json:1]: i checked in a blank one so that the typescript references work

Comment (bhansell1) [reply amplify_outputs.json:1]: I think we should add it to .gitignore and just have the file locally so it doesn't break. maybe something to add to the documentation? Or add creating the file to the `make config` command?

Comment (bhansell1) [reply amplify_outputs.json:1]: Or you could use fs to read in the outputs file and return an object like so;

```
import fs from 'node:fs';

export const config = () => {
  try {
    const output = fs.readFileSync('../../amplify_outputs.json', 'utf8');
    return JSON.parse(output);
  } catch {
    return {};
  }
};
```

Comment (chris-elliott-nhsd) [reply amplify_outputs.json:1]: done - great idea!

**Review by bhansell1 — APPROVED** [review]

**Review by mark-r-bjss — DISMISSED** [review]

**Review by m-salaudeen — DISMISSED** [review]

**Review by m-salaudeen — APPROVED** [review]

**Pull request merged**